tools/mo/openvino/tools/mo/middle/ConcatOptimization.py

```diff
@@ -1,116 +0,0 @@
-                for x in arr:
-                    if used[concat_id][x]:
-                        we_can = False
-                        break
-
-                if not we_can:
-                    continue
-
-                for x in arr:
-                    used[concat_id][x] = True
-
-                edge_attrs = graph.get_edge_data(arr[0], concat_id)[0]
-                for in_node in arr:
-                    graph.remove_edge(in_node, concat_id)
-
-                new_input = mp[arr][1]
-                out_port = len(Node(graph, new_input).out_nodes()) + 1
-                edge_attrs['out'] = out_port
-                graph.add_edge(new_input, concat_id, **edge_attrs)
-
-                # Renumber 'in' attrs
-                concat_node = Node(graph, concat_id)
-                ln = len(concat_node.in_nodes())
-                ports = [x for x in concat_node.in_nodes().keys()]
-                ports.sort()
-
-                p_id = 0
-                for p in ports:
-                    in_node = concat_node.in_nodes()[p]
-                    graph[in_node.id][concat_id][0]['in'] = p_id
-                    p_id += 1
-
-
-class ConcatOdInputEraserAndPortsReconnect(MiddleReplacementPattern):
-    """
-    The transformation performs two actions with Concat operations:
-    1. Disconnects empty inputs (input tensor has at least one input dimension equal to 0)
-    2. Renumber Concat inputs to be 0, 1, 2,...
-    """
-    enabled = True
-    force_clean_up = True
-
-    def find_and_replace_pattern(self, graph: Graph):
-        for concat in graph.get_op_nodes(type='Concat'):
-            for in_port in concat.in_ports().values():
-                if not in_port.disconnected():
-                    shape = in_port.data.get_shape()
-                    assert shape is not None
-                    if 0 in shape:
-                        concat.delete_input_port(in_port.idx)
-
-            connected_ports = [port for port_idx, port in sorted(concat.in_ports().items()) if not port.disconnected()]
-            assert len(connected_ports), 'Concat "{}" have no inputs after removing inputs with 0 dimensions' \
-                                         ''.format(concat.soft_get('name', concat.id))
-
-            max_port_index = max([port_idx for port_idx in concat.in_ports().keys()])
-            # re-connect input ports sequentially and remove all not used
-            port_idx_to_connect = 0
-            for port_idx in range(max_port_index + 1):
-                if concat.is_in_port_connected(port_idx):
-                    if port_idx != port_idx_to_connect:
-                        concat.add_input_port(port_idx_to_connect, skip_if_exist=True)
-                        concat.in_port(port_idx).get_connection().set_destination(concat.in_port(port_idx_to_connect))
-                    port_idx_to_connect += 1
-                elif port_idx in concat.in_ports():
-                    concat.delete_input_port(port_idx)
```

Comment:
why did you remove this transformation?

Reply:
Thanks for noticing! I removed by mistake `ConcatOptimization` is not used but `ConcatOdInputEraserAndPortsReconnect` is used. You are right. Recovered it